balena/models/device_os.py

```diff
@@ -277,7 +278,7 @@ def get_supported_versions(self, device_type):
 
         response = self.base_request.request(
             '/device-types/v1/{device_type}/images'.format(device_type=device_type), 'GET',
-            endpoint=self.settings.get('api_endpoint')
+            endpoint=self.settings.get('api_endpoint'), auth=auth
```

Comment:
@thgreasi, the `BaseRequest` class does check for auth token or user api key and automatically add it to the request headers if users don't explicitly set `auth=False` in some methods support that argument so I think it's doing what you mentioned.

I think provide `auth` argument here just allows users to get less data and send fewer requests to the API if they want since not all users need data about private DTs and ignoring authentication means we can skip the check for auth token expired time (sometimes we need to send a request to refresh token) on each base_request request.

Reply:
@nghiant2710 in that case, can we default auth to None, so that when the user doesn't provide a specific option, BaseRequest does "if logged in then pass the token" trick? That's how the the js sdk does it as well. So it's like the method works depending on the state of the sdk instance the user is calling it on.

To be clear, feel free to self-certify, since that's already covering the use case we cared about.